### PR TITLE
Dump stored items in crafting reagents

### PR DESCRIPTION
**Pull request by chromiumboy**
## About the PR

This PR attempts to resolve issue #10852. When a storage container is used as a reagent for crafting, the items stored inside of it are dumped on the ground.

Attempting to craft a medibot with a full medkit (sans health analyzer)
medibot-crafting1

All contents of the medkit are dumped the ground when crafting starts
medibot-crafting2

**Media**

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- fix: Containers used in recipes now drop their contents when crafted.

### CODE REVIEW

File: Content.Server/Construction/ConstructionSystem.Initial.cs [hunk 1]
```diff
@@ -203,16 +207,31 @@ void ShutdownContainers()
                             if (used.Contains(entity))
                                 continue;
 
+                            // Dump out any stored entities in used entity
+                            if (TryComp<SharedStorageComponent>(entity, out var storage) && storage.StoredEntities != null)
+                            {
+                                foreach (var storedEntity in storage.StoredEntities.ToList())
+                                {
+                                    var transform = Transform(storedEntity);
```
Comment (Partmedia) [inline Content.Server/Construction/ConstructionSystem.Initial.cs:215]: Could you help me double check that replacing lines 215-219 with a call to `Content.Server/Storage/EntitySystems/StorageSystem.cs:RemoveAndDrop` does the same thing? It looks like it correctly handles some more book-keeping logic related to storage systems.

File: Content.Server/Construction/ConstructionSystem.Initial.cs [hunk 2]
```diff
@@ -203,16 +207,31 @@ void ShutdownContainers()
...
+                                    storage.Remove(storedEntity);
+                                }
+                            }
+
                             if (string.IsNullOrEmpty(arbitraryStep.Store))
                             {
                                 if (!container.Insert(entity))
                                     continue;
                             }
+
```
Comment (Partmedia) [inline Content.Server/Construction/ConstructionSystem.Initial.cs:228]: Were these whitespace changes intended?

**Review by Partmedia — COMMENTED** [review]
Thank you for your investigation and fix.

File: Content.Server/Construction/ConstructionSystem.Initial.cs (hunk 1 above)
Comment (chromiumboy) [reply Content.Server/Construction/ConstructionSystem.Initial.cs:221]: Good find. This has been integrated

File: Content.Server/Construction/ConstructionSystem.Initial.cs (hunk 2 above)
Comment (chromiumboy) [reply Content.Server/Construction/ConstructionSystem.Initial.cs:232]: The white spacing seem fine to me

**Comment by chromiumboy** [comment]
Hi @Partmedia, your questions have been addressed.

File: Content.Server/Construction/ConstructionSystem.Initial.cs (hunk 2 above)
Comment (Partmedia) [reply Content.Server/Construction/ConstructionSystem.Initial.cs:232]: This white space is extraneous because it separates the else from the main body of the if. But don't worry about fixing this, I will fix this for you.

File: Content.Server/Construction/ConstructionSystem.Initial.cs [hunk 3]
```diff
@@ -203,16 +211,31 @@ void ShutdownContainers()
...
                                 continue;
 
+                            // Dump out any stored entities in used entity
+                            if (TryComp<ServerStorageComponent>(entity, out var storage) && storage.StoredEntities != null)
+                            {
+                                foreach (var storedEntity in storage.StoredEntities.ToList())
+                                {
+                                    _storageSystem.RemoveAndDrop(entity, storedEntity, storage);
+
+                                    var transform = Transform(storedEntity);
```
Comment (Partmedia) [inline Content.Server/Construction/ConstructionSystem.Initial.cs:221]: Transform reparenting is already handled by `RemoveAndDrop`. Orientation randomization should probably be moved into `RemoveAndDrop`. Since that touches unrelated code, I'm just going to remove it from here and add the randomization separately.

**Comment by Partmedia** [comment]
Thank you for your contribution. Will merge (with some modifications) once the checks pass.
